### PR TITLE
Update migration guide.

### PR DESCRIPTION
**Pull request by boczkowska**
Added description of v2 changes w/ links to respective issues and some language suggestions. 

Docs issue: https://github.com/devfile/api/issues/160

### CODE REVIEW

**Comment by boczkowska** [comment]
@elsony could you PTAL if everything is correct content-wise?
@jc-berger could you please provide a language review? :relaxed: 

Thanks.

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 1]
```diff
@@ -175,27 +175,27 @@ components:
 
 Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/31[corresponding issue].
```
Comment (elsony) [inline docs/modules/ROOT/pages/migration_guide.adoc:176]: Change URL prefix, see above.

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 2]
```diff
@@ -175,27 +175,27 @@ components:
...
 
-=== Apply Command
+=== Applying command
 
 Details can be found in the https://github.com/devfile/api/issues/56[corresponding issue].
 
-=== Environment Varibables for a Specific Command
+=== Environment variables for a specific command
 
 Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/21[corresponding issue].
```
Comment (elsony) [inline docs/modules/ROOT/pages/migration_guide.adoc:192]: Change URL prefix, see above.

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 3]
```diff
@@ -175,27 +175,27 @@ components:
...
 
-=== Renaming workdir into workingDir
+=== Renaming `workdir` into `workingDir`
 
 Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/22[corresponding issue].
```
Comment (elsony) [inline docs/modules/ROOT/pages/migration_guide.adoc:196]: Change URL prefix, see above.

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 4]
```diff
@@ -175,27 +175,27 @@ components:
...
 
-=== Id and label for Composite Commands
+=== `Id` and `label` for composite commands
 
 Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/18[corresponding issue].
```
Comment (elsony) [inline docs/modules/ROOT/pages/migration_guide.adoc:200]: Change URL prefix, see above.

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 5]
```diff
@@ -288,5 +288,99 @@ Details are in the https://github.com/che-incubator/devworkspace-api/issues/32[c
...
+            value: "John Doe"
+        commandLine: 'echo "Hello ${USER}"'
+        component: build-tools
+----
+|
+[source,yaml]
+----
+v2.0
+---
+  - id: helloWorld
```
Comment (elsony) [inline docs/modules/ROOT/pages/migration_guide.adoc:315]: This should be `name` instead of `id`

**Review by elsony — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 6]
```diff
@@ -31,11 +31,11 @@ metadata:
...
 Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/7[corresponding issue].
 
-== How to migrate projects
+== Migrating projects
 
-There are NO major changes in the `projects` section of the devfile. A project specified in a v1.0 devfile will work as it is in a devfile v2.0. The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user pick one) and the git repo get not be cloned but only source code get copied (kind of what `git archive --remote=<repository URL> | tar -t` would do).
+A project specified in a v1.0 devfile works as it is in a devfile v2.0. The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user must pick one) and the Git repo cannot be cloned but only source code get copied (kind of what `git archive --remote=<repository URL> | tar -t` would do).
 
-=== starterProjects and projects description
+=== `starterProjects` and projects description
```
[the remark this answers is not in the document]
Comment (boczkowska) [reply docs/modules/ROOT/pages/migration_guide.adoc]: Good catch! Added backticks around `project.`

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 7]
```diff
@@ -50,11 +50,11 @@ starterProjects:
 
 Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/42[corresponding issue].
 
-== How to migrate components
+== Migrating components
 
-There ARE major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will NOT work as it is in a devfile v2.0.
+A component specified in a v1.0 devfile will not work as it is in a devfile v2.0.
```
[the remark this answers is not in the document]
Comment (boczkowska) [reply docs/modules/ROOT/pages/migration_guide.adoc]: I've readded the sentence. Probably removed it by accident the first time.

**Comment by boczkowska** [comment]
@elsony Thank you for the review! I've changed the URLs and addressed your other feedback. PTAL when you have time.

**Review by elsony — APPROVED** [review]
lgtm

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 8]
```diff
@@ -73,37 +72,37 @@ components:
...
-Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/14[corresponding issue].
+Details can be found in the https://github.com/devfile/api/issues/14[corresponding issue].
 
 === Containers endpoints (routes/ingresses)
 
-Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/33[corresponding issue].
+Details can be found in the https://github.com/devfile/api/issues/33[corresponding issue].
 
-== How to migrate plugins
+== Migrating plugins
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:105]:
```suggestion
== Migrating plug-ins
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 9]
```diff
@@ -73,37 +72,37 @@ components:
...
 
 There ARE major changes about the definition of plugins in a devfile. Plugins are now specified using a devfile. A plugin specified in a v1.0 meta.yaml will NOT work as it is in a devfile v2.0.
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:107]:
```suggestion
Plug-ins are specified using a devfile. A plug-in specified in a v1.0 `meta.yaml` does not work as is in a devfile v2.0.
```
Do not use CAPS in docs.

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 10]
```diff
@@ -73,37 +72,37 @@ components:
        reference: https://.../mongo.yaml
 ----
 
-Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/4[corresponding issue].
+Details can be found in the https://github.com/devfile/api/issues/4[corresponding issue].
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:75]: All of the _"Details can be found in the >corresponding issue<."_ sentences would look better with:

* No passive tense
* Title of the linked issue

E.g.: _"For more details, see >Polymorphic component type UX in a devfile<."_

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 11]
```diff
@@ -73,37 +72,37 @@ components:
...
-Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/19[corresponding issue].
+Details can be found in the https://github.com/devfile/api/issues/19[corresponding issue].
 
-=== Out of Main Pod Compoenents
+=== Out of main pod components
 
 Details can be found in the  https://github.com/devfile/api/issues/48[corresponding issue].
 
-=== Replace Alias with Name
+=== Replace alias with Name
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:85]:
```suggestion
=== Replace alias with name
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 12]
```diff
@@ -48,13 +48,12 @@ starterProjects:
...
-== How to migrate components
+== Migrating components
+There are major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will not work as it is in a devfile v2.0.
 
-There ARE major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will NOT work as it is in a devfile v2.0.
-
-=== Component is a polymophic type
+=== Component is a polymorphic type
 
 For a better syntax validation component now is defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:58]:
```suggestion
For better syntax validation, a component now is defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin`, or `volume`.
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 13]
```diff
@@ -282,11 +281,105 @@ events:
     - "openFile"
 ----
 
-Details are in the https://github.com/che-incubator/devworkspace-api/issues/32[corresponding issue].
+Details are in the https://github.com/devfile/api/issues/32[corresponding issue].
 
 == New metadata
 
-In v2 of the devfile we have added some new metadata like `version` (this is the https://github.com/che-incubator/devworkspace-api/issues/10[corresponding issue]) and some mandatory metadata for plugins (this is the https://github.com/devfile/api/issues/31[corresponding issue]).
+In v2 of the devfile we have added some new metadata like `version` (this is the https://github.com/devfile/api/issues/10[corresponding issue]) and some mandatory metadata for plugins (this is the https://github.com/devfile/api/issues/31[corresponding issue]).
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:288]:
```suggestion
Version 2 of the devfile adds new metadata, including `version` (see link:https://github.com/devfile/api/issues/10[The devfile schema/samples should have version and name]) and some mandatory metadata for plug-ins (see link:https://github.com/devfile/api/issues/31[Better plugin mechanism]).
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 14]
```diff
@@ -282,11 +281,105 @@ events:
...
+    git:
+      remotes:
+        origin: "https://github.com/amisevsk/web-nodejs-sample.git"
+        upstream: "https://github.com/che-samples/web-nodejs-sample.git"
+        gh-collaborator: "https://github.com/gh-collaborator/web-nodejs-sample.git"  
+      checkoutFrom: <1> 
+        revision: foo <2> 
+        remote: upstream <3> 
+----
+<1> Mandatory if there is more than one remote
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:343]:
```suggestion
<1> Mandatory if there is more than one remote.
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 15]
```diff
@@ -282,11 +281,105 @@ events:
...
+<2> The revision that should be used checked out. May be a branch name, tag or commit id. Default branch should be checked out if missing.
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:344]:
```suggestion
<2> The revision that should be used checked out. May be a branch name, tag, or commit id. Default branch should be checked out if missing.
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 16]
```diff
@@ -282,11 +281,105 @@ events:
...
+<3> The remote name should be used as init. Required if there are more than one remote configured.
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:345]:
```suggestion
<3> The remote name that should be used as init. Required if there are more than one remote configured.
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 17]
```diff
@@ -282,11 +281,105 @@ events:
...
+components:
+  - name: tools
+    container:
+----
+|====
+
+More details are in the https://github.com/devfile/api/issues/102[corresponding issue].
+
+== `mountSources` are true by default
+`mountSources` field is defined as `true` for all non-plugin components and components that do not set dedicatedPod to `true`.
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:375]:
```suggestion

The `mountSources` field is defined as `true` for all non-plug-in components and components that do not set `dedicatedPod` to `true`.
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 18]
```diff
@@ -282,11 +281,105 @@ events:
...
+More details are in the https://github.com/devfile/api/issues/75[corresponding issue].
 
+== `hotReloadCapable` added to `exec` commands
+`hotReloadCapable` defines whether the command is capable to reload itself when source code changes. If set to `true` the command will not be restarted and it is expected to handle file changes on its own.
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:379]:
```suggestion

The `hotReloadCapable` fields defines whether the command is capable of reloading itself when source code changes. If set to `true`, the command is not restarted, and it is expected to handle file changes on its own. 
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 19]
```diff
@@ -1,9 +1,9 @@
-= Devfile v2 Migration Guide
+= Migrating from devfile v1 to devfile v2
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:1]: Unless there's a specific reason not to, I believe the use of `v1` vs `v1.0` and `v2` vs `v2.0` should be unified.

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 20]
```diff
@@ -1,9 +1,9 @@
...
 :description: A guide to migrate a v1 Devfile to v2
 
 This guide explains how to migrate an existing v1.0 devfile to v2.0.
 
-== Schema Version
+== Schema version
 
 Attribute `apiVersion` has been renamed `schemaVersion`:
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:8]:
```suggestion
Attribute `apiVersion` has been renamed to `schemaVersion`:
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 21]
```diff
@@ -29,13 +29,13 @@ metadata:
...
 |====
 
-Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/7[corresponding issue].
+Details can be found in the https://github.com/devfile/api/issues/7[corresponding issue].
 
-== How to migrate projects
+== Migrating projects
 
-There are NO major changes in the `projects` section of the devfile. A project specified in a v1.0 devfile will work as it is in a devfile v2.0. The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user pick one) and the git repo get not be cloned but only source code get copied (kind of what `git archive --remote=<repository URL> | tar -t` would do).
+A project specified in a v1.0 devfile works as it is in a devfile v2.0. The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user must pick one) and the Git repo cannot be cloned but only source code get copied (kind of what `git archive --remote=<repository URL> | tar -t` would do).
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:36]:
```suggestion
A project specified in a v1.0 devfile works as is in a devfile v2.0. The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user must pick one) and the Git repository cannot be cloned. Instead, only source code is to be copied (for example, `git archive --remote=_<repository URL>_ | tar -t`).
```

File: docs/modules/ROOT/pages/migration_guide.adoc [hunk 22]
```diff
@@ -48,13 +48,12 @@ starterProjects:
     description: "Use this app to get a simple "hello world" nodejs application"
 ----
 
-Details can be found in the https://github.com/che-incubator/devworkspace-api/issues/42[corresponding issue].
+Details can be found in the https://github.com/devfile/api/issues/42[corresponding issue].
 
-== How to migrate components
+== Migrating components
+There are major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will not work as it is in a devfile v2.0.
```
Comment (rkratky) [inline docs/modules/ROOT/pages/migration_guide.adoc:54]:
```suggestion

There are major changes in the `components` section of the devfile. A component specified in a v1.0 devfile does not work as is in a devfile v2.0.
```

**Review by rkratky — CHANGES_REQUESTED** [review]
@boczkowska, adding a bunch of nitpicks.

**Comment by boczkowska** [comment]
@rkratky thanks! PTAL

**Review by rkratky — APPROVED** [review]
Thanks, @boczkowska.

**Pull request merged**